pysteps/io/importers.py

```diff
@@ -1617,3 +1621,384 @@ def _import_saf_crri_geodata(filename):
+
+    Parameters
+    ----------
+    filename: str
+        Name of the file to import.
+    product: {'WX','RX','EX','RY','RW','AY','RS','YW','WN'}
+        The specific product to read from the file. Please see
+        https://www.dwd.de/DE/leistungen/radolan/radolan_info/
+        radolan_radvor_op_komposit_format_pdf.pdf
+        for a detailed description.
+
+    {extra_kwargs_doc}
+
+    Returns
+    -------
+    out: tuple
+        A three-element tuple containing the precipitation field in mm/h imported
+        from a MeteoSwiss gif file and the associated quality field and metadata.
+        The quality field is currently set to None.
+    """
+    size_file = os.path.getsize(filename)
+    size_data = np.round(size_file, -3)
+    size_header = size_file - size_data
+
+    f = open(filename, "rb")
+    header = f.read(size_header).decode("utf-8")
+
+    prod = header[:2]
+
+    assert prod == product, "Product not in File!"
+
+    prod_cat1 = np.array(["WX", "RX"])
+    prod_cat2 = np.array(["RY", "RW", "YW"])
+
+    nbyte = 1 if prod in prod_cat1 else 2
+    signed = "B" if prod in prod_cat1 else "H"
+
+    fac = int(header.split("E-")[1].split("INT")[0])
+    dimsplit = header.split("x")
+    dims = np.array((dimsplit[0][-4:], dimsplit[1][:4]), dtype=int)[::-1]
+
+    data = array.array(signed)
+    data.fromfile(f, size_data // nbyte)
+
+    f.close()
+
+    data = np.array(np.reshape(data, dims, order="F"), dtype=float).T
+
+    if prod == "SF":
+        no_echo_value = 0.0
+    elif prod in prod_cat2:
+        no_echo_value = -0.01
+    else:
+        no_echo_value = -32.5
+
+    if prod in prod_cat1:
+        data[data >= 249] = np.nan
+        data = data / 2.0 + no_echo_value
+    elif prod in prod_cat2:
+        data, no_data_mask = _identify_info_bits(data)
+        if prod == "AY":
+            data = (10 ** (fac * -1)) * data / 2.0 + no_echo_value
+        else:
+            data = (10 ** (fac * -1)) * data + no_echo_value
+    else:
+        data, no_data_mask = _identify_info_bits(data)
+        data = (10 ** (fac * -1)) * data / 2.0 + no_echo_value
+
+    data[no_data_mask] = np.nan
+
+    geodata = _import_dwd_geodata(product, dims)
+    metadata = geodata
+
+    return data, None, metadata
+
+
+def _identify_info_bits(data):
+
+    # clutter
+    clutter_mask = data - 2**15 >= 0.0
+    data[clutter_mask] = 0
+
+    # negative values
+    mask = data - 2**14 >= 0.0
+    data[mask] -= 2**14
+
+    # no data
+    no_data_mask = data - 2**13 == 2500.0
+
+    if np.sum(no_data_mask) == 0.0:
+
+        no_data_mask = data - 2**13 == 0.0
+
+    data[no_data_mask] = 0.0
+
+    # secondary data
+    data[data - 2**12 > 0.0] -= 2**12
+
+    data[mask] *= -1
+
+    return data, no_data_mask
+
+
+def _import_dwd_geodata(product, dims):
+    """
+    Geodata for RADOLAN products. Hard-coded here, cause there is only basic
+    information about the projection in the header of the binary RADOLAN
+    files.
+    """
```

Comment:
Add some more docstring here, too (input parameters and output). I have already added much more docstrings, but feel free to edit them from here.